radixdlt-core/radixdlt/src/test/java/com/radixdlt/network/p2p/proxy/PeerProxyTest.java

```diff
@@ -0,0 +1,215 @@
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.network.p2p.proxy;
+
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.counters.SystemCounters.CounterType;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.network.p2p.test.DeterministicP2PNetworkTest;
+import com.radixdlt.network.p2p.test.P2PTestNetworkRunner.NodeProps;
+import com.radixdlt.networks.Addressing;
+import com.radixdlt.networks.Network;
+import org.junit.After;
+import org.junit.Test;
+import com.radixdlt.network.p2p.liveness.messages.PeerPingMessage;
+
+import java.time.Duration;
+import java.util.Set;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public final class PeerProxyTest extends DeterministicP2PNetworkTest {
+
+	private static final int VALIDATOR_NODE = 0;
+	private static final int PROXY_NODE = 1;
+	private static final int EXTERNAL_NODE = 2;
+
+	@After
+	public void cleanup() {
+		testNetworkRunner.cleanup();
+	}
+
+	@Test
+	public void test_peer_proxy() throws Exception {
```

Comment:
Nice test, and thanks for the comments throughout to clearly mark what's happening!